tests/integration/pilot/cni/e2e-taint/e2e_taint_test.go

```diff
@@ -0,0 +1,284 @@
+				fileLocation       string
+			}
+			type args struct {
+				config             configLocation
+				daemonsetLocations []string
+				namespaces         []v1.Namespace
+			}
+			tests := []struct {
+				name string
+				args args
+			}{{
+				name: "single case",
+				args: args{
+					config:             configLocation{fileLocation: "testdata/singleConfig", configmapNamespace: "default", configmapName: "single"},
+					daemonsetLocations: []string{"testdata/podsConfig/single-critical-label.yaml"},
+					namespaces:         []v1.Namespace{Test1NS},
+				},
+			},
+				{
+					name: "multiple labels in two namespace",
+					args: args{
+						config:             configLocation{fileLocation: "testdata/multiConfig", configmapNamespace: "default", configmapName: "multi"},
+						daemonsetLocations: []string{"testdata/podsConfig/single-critical-label.yaml", "testdata/podsConfig/addon-test2-in-test2.yaml"},
+						namespaces:         []v1.Namespace{Test1NS, Test2NS},
+					},
+				},
+				{
+					name: "multiple labels in one namespace",
+					args: args{
+						config:             configLocation{fileLocation: "testdata/multiInOneConfig", configmapNamespace: "default", configmapName: "mutli.one"},
+						daemonsetLocations: []string{"testdata/podsConfig/single-critical-label.yaml", "testdata/podsConfig/addon-test1.yaml"},
+						namespaces:         []v1.Namespace{Test1NS},
+					},
+				},
+			}
+			for _, tt := range tests {
+				ctx.NewSubTest(tt.name).Run(func(ctx framework.TestContext) {
+					clientSet, err := clientSetup()
+					if err != nil {
+						t.Fatalf(err.Error())
+					}
+					for _, ns := range tt.args.namespaces {
+						_, err = clientSet.CoreV1().Namespaces().Create(context.TODO(), &ns, metav1.CreateOptions{})
+						if err != nil {
+							t.Fatalf(err.Error())
+						}
+						err = retry.UntilSuccess(func() error {
+							if kube.NamespaceExists(clientSet, ns.Name) {
+								return nil
+							}
+							return fmt.Errorf("cannot create ns %s", ns.Name)
+						}, retry.Timeout(TimeDuration), retry.Delay(TimeDelay))
+						if err != nil {
+							t.Fatalf(err.Error())
+						}
+					}
+					TaintOptions := &taint.Options{
+						ConfigmapName:      tt.args.config.configmapName,
+						ConfigmapNamespace: tt.args.config.configmapNamespace,
+					}
+					_, err = buildConfigIfNotExists(clientSet, *TaintOptions, tt.args.config.fileLocation)
+					if err != nil {
+						t.Fatalf(err.Error())
+					}
+					taintSetter, err := taint.NewTaintSetter(clientSet, TaintOptions)
+					if err != nil {
+						log.Fatalf("Could not construct taint setter: %s", err)
+					}
+					tc, err := taint.NewTaintSetterController(taintSetter)
+					if err != nil {
+						log.Fatalf("Could not build controller %s", err)
+					}
+					stopCh := make(chan struct{})
+					go tc.Run(stopCh)
+					tc.RegistTaints()
+					//here all nodes have readiness taint
+					err = retry.UntilSuccess(func() error {
+						for _, node := range tc.ListAllNode() {
```

Comment:
this is returning empty list. The test is not really testing anything, which makes sense since it should not pass - the daemonsets we are running are invalid (bad image)